src/app/server/Mdns.cpp

```diff
@@ -64,6 +65,26 @@ NodeId GetCurrentNodeId()
     return chip::kTestDeviceNodeId;
 }
 
+// Requires an 8-byte mac to accommodate thread.
+chip::ByteSpan FillMAC(uint8_t mac[8])
+{
+    memset(mac, 0, 8);
+    if (DeviceLayer::ConfigurationMgr().GetPrimary802154MACAddress(mac) == CHIP_NO_ERROR)
```

Comment:
I'm a little bit confused here. A few days ago #5343 was merged which added support for setting hostname. Please look [here](CHIP_ERROR DiscoveryImplPlatform::SetupHostname()). Is it a different implementation?

Side note: The specification defines the usage of EUI-64 for Thread, so we should use `_GetFactoryAssignedEUI64` instead.

Reply:
Crossed streams a bit - I didn't see that go in. But without this change, the hostname I'm getting back for non-thread devices is definitely incorrect even with #5343. Let me see if I can merge these two implementations a bit more nicely

Reply:
That's true, WiFi implementation from #5343 may not be correct, as I didn't modify existing solution. I just added missing part of creating hostname for Thread devices.